scripts/zones/Aht_Urhgan_Whitegate/npcs/Shajaf.lua

```diff
@@ -8,13 +8,55 @@ entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    player:startEvent(162)
+    local imperialStanding = player:getCurrency('imperial_standing')
+
+    -- Players can only purchase a KI once every real-life day.
+    -- This var is set onEventFinish
+    if player:getCharVar('isnmAccepted') == 1 then
+        -- 'I don't have any offers for you at the moment'
+        player:startEvent(163)
+    elseif
+        player:hasKeyItem(xi.ki.CONFIDENTIAL_IMPERIAL_ORDER) or
+        player:hasKeyItem(xi.ki.SECRET_IMPERIAL_ORDER)
+    then
+        -- 'What are you doing, running off from a job?'
+        player:startEvent(161, 0)
+    elseif
+        player:hasKeyItem(xi.ki.PSC_WILDCAT_BADGE) and
+        imperialStanding >= 2000 -- Prevent against packet injection
+    then
+        --TODO: It is unclear what params 3-9 affect.
+        player:startEvent(160, imperialStanding, 1, 1791, 0, 0, 0, 2, 4095)
```

Comment:
![imagen](https://github.com/LandSandBoat/server/assets/60053999/50f00089-643a-46f9-9218-61b8f40cbcd6)

As you can see, only event ID and the first 2 event params are needed. So:
```lua
-- Parameter 2 unknown. Parameters 3-8 do nothing,
player:startEvent(160, imperialStanding, 1)
```